lib/map.h

```diff
@@ -17,42 +17,66 @@ limitations under the License.
 #ifndef LIB_MAP_H_
 #define LIB_MAP_H_
 
-#include <map>
+#include <optional>
 
-template <class K, class T, class V, class Comp, class Alloc>
-inline V get(const std::map<K, V, Comp, Alloc> &m, T key, V def = V()) {
+/// Given a map and a key, return the value corresponding to the key in the map,
+/// or a given default value if the key doesn't exist in the map.
+template <typename Map, typename Key>
+typename Map::mapped_type get(const Map &m, const Key &key) {
     auto it = m.find(key);
-    if (it != m.end()) return it->second;
-    return def;
+    return it != m.end() ? it->second : typename Map::mapped_type{};
+}
+template <class Map, typename Key = typename Map::key_type,
+          typename Value = typename Map::mapped_type>
+typename Map::mapped_type get(const Map &m, const Key &key, Value &&def) {
```

Comment:
I think we could use a version that takes `const Value &def` for cases where the default is an lvalue and we want it copied when used.

Reply:
See above. I believe the universal reference version with `std::forward` should take care of this.